contracts/Staking.sol

```diff
@@ -165,6 +156,30 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
         _decreaseLockAmountUnsafe(_from, msg.sender, _amount);
     }
 
+    /**
+     * @notice Transfer `@tokenAmount(stakingToken: address, _allowance)` from `_from`'s lock by `msg.sender` to `_to` (unstaked)
+     * @param _from Owner of locked tokens
+     * @param _to Recipient of the tokens
+     * @param _amount Number of tokens to be transferred
+     */
+    function transferFromLockAndUnstake(
+        address _from,
+        address _to,
+        uint256 _amount
+    )
+        external
+        isInitialized
+    {
+        Account storage account = accounts[_from];
+        Lock storage lock = account.locks[msg.sender];
+        // check that lock is enough, it also means that lock.amount > 0 and therefore hasn't been unlocked
+        require(lock.amount >= _amount, ERROR_NOT_ENOUGH_LOCK);
+
+        _decreaseLockAmountUnsafe(_from, msg.sender, _amount);
+        _transfer(_from, msg.sender, _to, address(0), _amount);
+        _unstakeUnsafe(_to, _amount, new bytes(0));
```

Comment:
I think it sounds safer to first unstake and then transfer

Reply:
But transfer it’s not actually transferring (like token transfer), only modifying Staking balances, and receiver could not unstake without having balance. (This is the “natural” order).